### PR TITLE
[dbus] add "Detach" API

### PR DESCRIPTION
**Pull request by jinran-google**
Fixes <https://github.com/openthread/ot-br-posix/issues/755>.

### CODE REVIEW

File: src/agent/thread_helper.cpp [hunk 1]
```diff
@@ -292,6 +292,11 @@ void ThreadHelper::Attach(ResultHandler aHandler)
     }
 }
 
+otError ThreadHelper::Detach(void)
+{
+    return otThreadSetEnabled(mInstance, false);
```
Comment (zhanglongxia) [inline src/agent/thread_helper.cpp:297]: The "Attach" function calls the function `otIp6SetEnabled` to start the Ipv6 stack, should we  also call the `otIp6SetEnabled` to disable the ipv6 stack in the `Detach` function?

File: src/dbus/common/constants.hpp [hunk 2]
```diff
@@ -46,6 +46,7 @@
 
 #define OTBR_DBUS_SCAN_METHOD "Scan"
 #define OTBR_DBUS_ATTACH_METHOD "Attach"
+#define OTBR_DBUS_DETACH_METHOD "Detach"
```
Comment (bukepo) [inline src/dbus/common/constants.hpp:49]: update https://github.com/openthread/ot-br-posix/blob/main/src/dbus/server/introspect.xml?

Comment (jinran-google) [reply src/dbus/common/constants.hpp:49]: It's already updated <https://github.com/openthread/ot-br-posix/pull/767/files#diff-64258c71b54a165182be9a632cb0e131147fa82efdce34730e5a5c1b40c58c3bR49-R52>

File: src/agent/thread_helper.cpp (hunk 1 above)
Comment (jinran-google) [reply src/agent/thread_helper.cpp]: Updated. I think both are OK, but disabling the stack may save some resources.

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**